recipes/libtiff/all/conanfile.py

```diff
@@ -44,14 +44,25 @@ def _source_subfolder(self):
     def _build_subfolder(self):
         return "build_subfolder"
 
+    @property
+    def _has_web_option(self):
+        return tools.Version(self.version) >= "4.0.10"
+
+    @property
+    def _has_zstd_option(self):
+        return tools.Version(self.version) >= "4.0.10"
+
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
-        if tools.Version(self.version) < "4.1.0":
+        if not self._has_web_option:
             del self.options.webp
+        if not self._has_zstd_option:
             del self.options.zstd
 
     def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
```

Comment:
I would not delete the fPIC option.

Reply:
Could you elaborate please?

Reply:
linux compilers need `fPIC`, the delete is to prevent packages which are unusable to a consumer

Reply:
@SpaceIm Doesn't the CMake helper already do that for you?

Reply:
deleting fPIC option is not about CMake helper function, but package_id, (fPIC=False and shared doesn't make sense).

It's a pattern used in all recipes (except old ones not yet updated)

Reply:
thanks for explanation. Should this pattern be used only for CMake recipes or also for all others? Who makes sure that the -fPIC option for the compiler/linker call is set correctly when no CMake is used?

Reply:
It depends. There is no silver bullet.

But basically, with meson, autotools and CMake helper, it should work.

If some custom build scripts are called, you may need to manually map a lot of profile settings and options with this script. Sometimes, injecting vars of `AutotoolsBuildEnvironment` in environment before calling those scripts is sufficient.